### PR TITLE
Adds perf tests for `Map.merge`.

### PR DESCRIPTION
**Pull request by abacaphiliac**
Output sample:
```
Map > merge an object of 2
  Old:   932,047   960,634   991,030 ops/sec
  New:   978,449   983,047   987,688 ops/sec
  compare: -1 1
  diff: 2.33%
  rme: 2.19%
Map > merge an object of 8
  Old:   568,422   572,977   577,606 ops/sec
  New:   571,550   575,883   580,283 ops/sec
  compare: 0 0
  diff: 0.5%
  rme: 0.78%
Map > merge an object of 32
  Old:   251,965   254,599   257,289 ops/sec
  New:   228,904   232,267   235,731 ops/sec
  compare: 1 -1
  diff: -8.78%
  rme: 1.27%
Map > merge an object of 1024
  Old:     8,428     8,491     8,554 ops/sec
  New:     7,625     7,681     7,739 ops/sec
  compare: 1 -1
  diff: -9.54%
  rme: 0.73%
Map > mergeDeep an object of 2
  Old:   321,103   323,956   326,860 ops/sec
  New:   323,939   328,831   333,872 ops/sec
  compare: -1 1
  diff: 1.5%
  rme: 1.23%
Map > mergeDeep an object of 8
  Old:   125,157   126,434   127,736 ops/sec
  New:   124,574   126,283   128,040 ops/sec
  compare: 0 0
  diff: -0.12%
  rme: 1.2%
Map > mergeDeep an object of 32
  Old:    37,218    37,439    37,662 ops/sec
  New:    36,882    37,395    37,923 ops/sec
  compare: 0 0
  diff: -0.12%
  rme: 1.06%
Map > mergeDeep an object of 1024
  Old:       728       740       752 ops/sec
  New:       698       712       726 ops/sec
  compare: 1 -1
  diff: -3.82%
  rme: 1.8%
```

Full output: [map-merge-perf.txt](https://github.com/facebook/immutable-js/files/1395673/map-merge-perf.txt)

### CODE REVIEW

**Comment by abacaphiliac** [comment]
@leebyron please review.

File: perf/Map.js [hunk 1]
```diff
@@ -139,4 +139,39 @@ describe('Map', function () {
...
+  describe('merge an object', function () {
+
+    [2, 8, 32, 1024].forEach(size => {
+      const obj = {};
+      for (let ii = 0; ii < size; ii++) {
+        obj['x' + ii] = ii;
+      }
+
+      const m1 = Immutable.Map(obj);
+      const m2 = Immutable.Map(obj);
```
Comment (leebyron) [inline perf/Map.js]: These two are the same - there's a special path for no-op merges, so you might want to add a separate perf test  where the second map contains different values

File: perf/Map.js [hunk 2]
```diff
@@ -139,4 +139,39 @@ describe('Map', function () {
 
   });
 
+  describe('merge an object', function () {
```
Comment (leebyron) [inline perf/Map.js]: `'merge a map'`?

**Comment by abacaphiliac** [comment]
@leebyron Done. Thanks!

Are you interested in me getting the perf tests running in Travis?

**Comment by abacaphiliac** [comment]
@leebyron Resolved (style) conflicts.

**Comment by leebyron** [comment]
I think the perf tests might be too slow to run in Travis, also they're pretty sensitive the the operating environment so I think they may also be too noisy

**Comment by leebyron** [comment]
Thanks! I added a few additional changes. I followed up with the change I mentioned before about ensuring the second map contains different values. I also removed `mergeDeep()` since it seemed broken (it wasn't calling `mergeDeep()`) and that's just a thin layer atop `merge()` anyhow, so from a perf testing point of view, it's probably duplicative

**Pull request merged**